locale/en/docs/guides/backpressuring-in-streams.md

```diff
@@ -0,0 +1,264 @@
+  
+  rl.close();
+});
+```
+
+
+There are different functions to transfer data from one process to another. In 
+Node.js, there is an internal built-in function called [`.pipe()`][]. There are
+[other packages][] out there you can use too! Ultimately though, at the basic 
+level of this process, we have two separate components: the _source_ of the 
+data and the _consumer_.
+
+In Node.js the source is a [`Readable`][] stream and the consumer is the 
+[`Writable`][] stream (both of these may be interchanged with a [`Duplex`][] 
+stream, but that is out-of-scope for this guide).
+
+## Too Much Data, Too Quickly
+
+There are instance where a [`Readable`][] stream might give data to the
+[`Writable`][] much too quickly --- much more than the consumer can handle! 
+
+When that occurs, the consumer will begin to queue all the chunks of data for 
+later consumption. The write queue will get longer and longer, and because of 
+this more data must be kept in memory until the entire process has completed.
+
+```javascript
+  var fs = require('fs');
+
+  var inputFile  = fs.createReadStream('REALLY_BIG_FILE.x');
+  var outputFile = fs.createWriteStream('REALLY_BIG_FILE_DEST.x');
+
+  // Secretly the stream is saying: "whoa, whoa! hang on, this is way too much!"
+  inputFile.pipe(outputFile);
+```
+
+
+This is why backpressure is important. If a backpressure system was not 
+present, the process would use up your system's memory, effectively slowing 
+down other processes, and monopolizing a large part of your system until 
+completion.
+
+This results in a few things:
+
+* Memory exhaustion
+* A very overworked garbage collector
+* Slowing down all other current processes
+
+## Memory Exhaustion
+
+
+
+## Garbage Collection
+
+
+
+## Overall Dip in System Performace
+
+
+
+## How Does Backpressure Resolve These Issues?
+
+The moment that backpressure is triggered can be narrowed exactly to the return 
+value of a `Writable`'s [`.write()`][] function. This return value is 
+determined by a few conditions, of course.
+
+In any scenario where the data buffer has exceeded the [`highWaterMark`][] or 
+the write queue is currently busy, [`.write()`][] will return `false`. 
+
+When a `false` value is returned, the backpressure system kicks in. It will 
+pause the incoming [`Readable`][] stream from sending any data and wait until
+the consumer is ready again.
+
+Once the the queue is finished, backpressure will allow data to be sent again.
+The space in memory that was being used will free itself up and prepare for the 
+next glob of data. 
+
+This effectively allows an fixed amount of memory to be used at any given
+time for a [`.pipe()`][] function. There will be no memory leakage, no 
+indefinite buffering, and the garbage collector will only have to deal with
+one area in memory!
+
+So, if backpressure is so important, why have you (probably) not heard of it? 
+Well the answer is simple: Node.js does all of this automatically for you. 
```

Comment:
Make it be known somewhere that only `.pipe()` does this automatically (`readable.on('data', buf => writable.write(buf));` doesn't use backpressure, for example). I know in the next section you are looking more closely at `.pipe()`, but it seems to be assumed that the reader knows all this is talking about `.pipe()`.